api/v14/api_v14_snapshots.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright (c) 2025 Dell Inc, or its subsidiaries.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v14
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/dell/goisilon/api"
+)
+
+// GetIsiWritableSnapshots retrieves a list of writable snapshots.
+//
+// ctx: the context.
+// client: the API client.
+// Returns a list of writable snapshots and an error in case of failure.
+func GetIsiWritableSnapshots(
+	ctx context.Context,
+	client api.Client,
+) ([]*IsiWritableSnapshot, error) {
+	var resp *IsiWritableSnapshotQueryResponse
+	err := client.Get(ctx, writableSnapshotPath, "", nil, nil, &resp)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get writable snapshots from array: %v", err)
+	}
+
+	result := make([]*IsiWritableSnapshot, 0, resp.Total)
+	result = append(result, resp.Writable...)
+
+	for {
+		if resp.Resume == "" {
+			break
+		}
+
+		query := api.OrderedValues{
+			{[]byte("resume"), []byte(resp.Resume)},
+		}
+
+		resp = nil
+		err = client.Get(ctx, writableSnapshotPath, "", query, nil, &resp)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get writable snapshots (query mode) from array: %v", err)
+		}
+
+		result = append(result, resp.Writable...)
```

Comment:
```suggestion
		result[i] = resp.Writable
		i++
```

Reply:
resp.Writable is a list of snapshots. We want to flatten the list for the caller.